Content.Shared/Movement/Pulling/Components/PullerComponent.cs

```diff
@@ -43,4 +43,49 @@ public sealed partial class PullerComponent : Component
 
     [DataField]
     public ProtoId<AlertPrototype> PullingAlert = "Pulling";
+
+    [DataField]
+    public Dictionary<GrabStage, short> PullingAlertSeverity = new()
+    {
+        { GrabStage.No, 0 },
+        { GrabStage.Soft, 1 },
+        { GrabStage.Hard, 2 },
+        { GrabStage.Suffocate, 3 },
+    };
+
+    [DataField, AutoNetworkedField]
+    public GrabStage GrabStage = GrabStage.No;
+
+    [DataField, AutoNetworkedField]
+    public GrubStageDirection GrabStageDirection = GrubStageDirection.Increase;
+
+    [AutoNetworkedField]
+    public TimeSpan NextStageChange;
+
+    [DataField]
+    public TimeSpan StageChangeCooldown = TimeSpan.FromSeconds(1.5f);
+
+    [DataField]
+    public Dictionary<GrabStage, float> EscapeChances = new()
+    {
+        { GrabStage.No, 1f },
+        { GrabStage.Soft, 0.7f },
+        { GrabStage.Hard, 0.4f },
+        { GrabStage.Suffocate, 0.1f },
+    };
+
+    [DataField]
+    public float SuffocateGrabStaminaDamage = 10f;
+
+    [DataField]
+    public float GrabThrowDamageModifier = 1f;
+
+    [ViewVariables]
+    public List<EntityUid> GrabVirtualItems = new();
+
+    [ViewVariables]
+    public Dictionary<GrabStage, int> GrabVirtualItemStageCount = new()
+    {
+        { GrabStage.Suffocate, 1 },
+    };
```

Comment:
Grabbing needs to be its own component + system, even if grabbing can cause pulling they're still handling 2 different things.